Returning light

Haden2 opened the pull request:
Didn't do the code to return the light when catching the enemy but I know how. My issue is I added a flare to the light to give it a source and to make it look nicer, but I cannot get the flare to turn back on when pressing Q

Haden2 committed:
Added lens flare to light.

Got it to turn off when the enemy reaches the player, and I also got it
to turn off when pressing q. Couldn't get it to turn back on though
before going to work. Feel free to edit with other flares. The welding
flare looked pretty cool too.

Haden2 committed:
Shifted light position

I cannot get the flare to turn back on when pressing q.

thomas-j-sell commented on Assets/Scripts/ToggleFlashlight.cs:
```diff
@@ -4,16 +4,22 @@
+			light.intensity = lightIntensity; //.75
+			lf.brightness = lensBright; //2
+		}
       else
```
This else block is missing brackets.

thomas-j-sell commented on Assets/Scripts/ToggleFlashlight.cs:
```diff
@@ -4,16 +4,22 @@
+		}
       else
-        light.intensity = 0; //If the light was on, turn it off.
+      	 	light.intensity = 0; //If the light was on, turn it off.
```
Only this line is part of the else.

thomas-j-sell commented on Assets/Scripts/ToggleFlashlight.cs:
```diff
@@ -4,16 +4,22 @@
       else
-        light.intensity = 0; //If the light was on, turn it off.
+      	 	light.intensity = 0; //If the light was on, turn it off.
+			lf.brightness = 0;
```
This line is out side of the if/else so it always runs.

thomas-j-sell commented:
Fixing the else block bracketing fixed the issue for me.  I like the idea of the lens flare, but it seems to be a bit much with it always on.  I suggest animating it when the light turns on and letting it fade away, or maybe just subdue it a bit.

Haden2 commented:
Ok, that makes a lot of sense. What happened was when I went to add to the if statement, it told me there was an issue because there was an unexpected else statement. I realized there were no brackets separating the if and else statement but I only added brackets to the if, not the else. That makes sense. 

Animating it will be no issue whatsoever because it is already animated. Change the fade speed from 100 to 1 and you've got it. I had it like that because I was looking at the light as the source instead of a flare per say. Try the welding flare and see if you prefer that one. Fading it away a bit would be a good idea. It is a bit much, but I also wanted a slight handicap when it came to the flashlight. I dont want it to be the best spectrum to see out of. Having the light on increases the chances of the light monster showing up. Cause like the other spectrums have their own handicaps, so I wanted to make the flashlight kind of have a flaw as well, you know? I'll mess around with it some unless you do it before hand.

Haden2 committed:
Attempted returning light fix

Ran into a glitch with the flare and collider. Also realized flaw in my
original plan because I am getting the light back as soon as the enemy
takes it. I need to either add a delay to the time it takes to return
light, or figure out NavMeshPathStatus.PathComplete so this cannot be
activated until the enemy gets to its path.

Haden2 committed:
Closer

Light turns off when player and enemy meets. Light turns back on when
player and enemy meet again and enemy waits (currently) 3 seconds before
chasing the player again. The total time will be increased to around 15
or 20 seconds. The problem is getting the whole thing to start over
again without duplicating code. Need to find a way to recall the section
void OnTriggerEnter.

Haden2 committed:
Light returned

Everything with the light and ememy 1 now works correctly. Player cannot
get the light back until 3 seconds have passed. They cannot catch up
with the enemy anyway. Once they get their light back, the enemy waits
10 seconds. In the game I am going to add an audio source to the little
girl. She is going to start counting like it is a game of hide and seek
and she is going to seek you. So this gives you a head start to get away
from her as she returns to normal speed.

Haden2 commented:
Ok, this is fixed

Haden2 committed:
Minor edits

Removed the rotating script. Changed the tiles to 40 so they appear
smaller (the main character looked really short with the big tiles). And
deactivated the flare for the time being.
